opal/mca/memory/patcher/memory_patcher_component.c

```diff
@@ -121,12 +116,11 @@ void *__mmap (void *start, size_t length, int prot, int flags, int fd, off_t off
 
 static void *(*original_mmap)(void *, size_t, int, int, int, off_t);
 
-static void *intercept_mmap(void *start, size_t length, int prot, int flags, int fd, off_t offset)
+static void *_intercept_mmap(void *start, size_t length, int prot, int flags, int fd, off_t offset)
 {
-    OPAL_PATCHER_BEGIN;
     void *result = 0;
 
-    if (prot == PROT_NONE) {
+    if ((flags & MAP_FIXED) && (start != NULL)) {
```

Comment:
BTW before this change, mmap(PROT_NONE) was calling release hook, and after this change - it doesn't. Is this the desired behavior?

Reply:
It should be okay because previously that entire section was #if 0 so mmap() wasn't being intercepted at all. The PROT_NONE was old code and was #if 0 after a decision a long time ago that mmap PROT_NONE didn't need to be trigger a cache deregistration. So when I re-added mmap() for this commit I just used the UCX code to decide what should be intercepted.

Reply:
ok, i guess if flags do not have MAP+FIXED, it will not overwrite an existing mapping anyway